bundle/edu.gemini.lchquery.servlet/src/main/scala/edu/gemini/lchquery/servlet/LchQueryFunctor.scala

```diff
@@ -0,0 +1,230 @@
+package edu.gemini.lchquery.servlet
+
+import java.io.Serializable
+import java.security.Principal
+import java.text.SimpleDateFormat
+import java.util
+import java.util.{Date, TimeZone}
+import java.util.logging.{Level, Logger}
+
+import edu.gemini.odb.browser._
+import edu.gemini.pot.sp.{ISPNode, ISPObservation, ISPProgram}
+import edu.gemini.pot.spdb.{DBAbstractQueryFunctor, IDBDatabaseService, IDBFunctor, IDBParallelFunctor}
+import edu.gemini.skycalc.{DDMMSS, HHMMSS}
+import edu.gemini.spModel.gemini.obscomp.SPSiteQuality
+import edu.gemini.spModel.obs.{ObsClassService, ObservationStatus}
+import edu.gemini.spModel.rich.shared.immutable._
+import edu.gemini.spModel.target.SPTarget
+import edu.gemini.spModel.target.env.TargetEnvironment
+import edu.gemini.spModel.too.Too
+
+import scala.collection.JavaConverters._
+import scala.util.{Failure, Success, Try}
+
+
+class LchQueryFunctor(queryType: LchQueryFunctor.QueryType,
+                      programParams: List[(LchQueryParam[ISPProgram], String)],
+                      observationParams: List[(LchQueryParam[ISPObservation], String)])
+  extends DBAbstractQueryFunctor with IDBParallelFunctor {
```

Comment:
`IDBParallelFunctor` is defunct and should be `@deprecated`.  It comes from the days when we were planning on distributing the ODB across multiple processes on multiple servers.  You can safely delete the `mergeResults` method below and just extend `IDBFunctor`.  Nobody is calling `mergeResults` in this code base.